lib/pluginDecorators.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const { kPluginMetadata } = require('./symbols')
+require('reflect-metadata')
```

Comment:
I'm -1 on adding any dependency that modifies global state. Can we do this without this library or with a work-around?

Reply:
We could store it at the plugin using a symbol.